newhelm/cache_helper.py

```diff
@@ -0,0 +1,87 @@
+# mypy: ignore-errors
+from abc import ABC, abstractmethod
+import json
+import os
+from pydantic import BaseModel
+from sqlitedict import SqliteDict  # type: ignore
+from typing import Any, Dict, Generic, Mapping, TypeVar, Optional, get_args
+
+from newhelm.sut import SUT
+from newhelm.typed_data import TypedData
+
+RequestType = TypeVar("RequestType", bound=BaseModel)
+ResponseType = TypeVar("ResponseType", bound=BaseModel)
+
+
+class CacheHelper(ABC, Generic[RequestType, ResponseType]):
+    @abstractmethod
+    def get_cached_response(self, request: RequestType) -> Optional[ResponseType]:
+        """Returns the SUT's cached response to a request."""
+
+    @abstractmethod
+    def update_cache(self, request: RequestType, response: ResponseType):
+        """Cache the SUT's response in-memory."""
+
+
+class SUTResponseCacheHelper(CacheHelper[RequestType, ResponseType]):
```

Comment:
I think we can drop the "Helper" from these classes and their instances: They are the `cache`.